Content.Server/Sports/BaseballBatSystem.cs

```diff
@@ -0,0 +1,82 @@
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<BaseballBatComponent, WideAttackEvent>(OnBatAttack);
+        }
+
+        public void OnBatAttack(EntityUid uid, BaseballBatComponent component, WideAttackEvent args)
+        {
+            args.Handled = true;
+
+            if (!EntityManager.TryGetComponent<MeleeWeaponComponent>(component.Owner, out var meleeWeaponComponent))
+                return;
+            if (_gameTiming.CurTime < meleeWeaponComponent.CooldownEnd)
+                return;
+
+            var location = EntityManager.GetComponent<TransformComponent>(args.User).Coordinates;
+            var dir = args.ClickLocation.ToMapPos(EntityManager) - location.ToMapPos(EntityManager);
+
+            var dirForceMultiplier = _random.NextFloat(component.WackForceMultiplierMin, component.WackForceMultiplierMax);
+
+            var hitStrength = _random.NextFloat(component.WackStrengthMin, component.WackStrengthMax);
+
+            //The melee system uses a collision raycast but apparently that doesnt work with items so im using GetEntitiesInArc
+            foreach (var entity in _entityLookupSystem.GetEntitiesInArc(location, meleeWeaponComponent.Range, dir.ToAngle(), meleeWeaponComponent.ArcWidth))
+            {
+                //Checking to see if the items are actually throwable
+                if (!EntityManager.HasComponent<ItemComponent>(entity))
+                    return;
+                if (EntityManager.TryGetComponent<TransformComponent>(entity, out var transformComponent) && transformComponent.Anchored)
+                    return;
+                if (!EntityManager.TryGetComponent<PhysicsComponent>(entity, out var physicsComponent))
+                    return;
+
+                if (EntityManager.HasComponent<ThrownItemComponent>(entity) || !component.OnlyHitThrown)
+                {
+
+                    var rand = _random.Next(1, component.FireballChance + 1); // Rolling to see if we'll get the fireball
+
+                    if (rand != component.FireballChance)
+                    {
+                        physicsComponent.Momentum = Vector2.Zero; //stopping the item so we get a clean throw
+                        _throwingSystem.TryThrow(entity, dir * dirForceMultiplier, hitStrength, uid);
+                        _audioSystem.Play(component.HitSound, Filter.Pvs(args.User), args.User, AudioParams.Default);
+                        return;
+                    }
+
+                    //just shoots a wizard fireball
+                    var fireball = Spawn("ProjectileFireball", EntityManager.GetComponent<TransformComponent>(entity).Coordinates);
+                    EntityManager.DeleteEntity(entity);
+                    _gunSystem.ShootProjectile(fireball, dir, args.User);
+                    _audioSystem.Play("/Audio/Effects/baseball-hit-extreme.ogg", Filter.Pvs(args.User), args.User, AudioParams.Default);
+                }
```

Comment:
Suss on this so see what another maintainer thinks.

At a minimum it needs to be using queries and the code should probably continue not return.